src/expression.js

```diff
@@ -73,6 +74,10 @@ var getKeyComputeData = function (key, scope, readOptions) {
 			return computeOrFunction();
 		}
 
+		if (computeOrFunction.computeData != null && computeOrFunction.hasOwnProperty("value")) {
```

Comment:
Ah, I see where the test as written changes from when I was probing at the parser from the console.  It doesn't hit this branch when you give CanMap/CanList data to the stache like `"<p>{{#each foo}}{{ bar[%index] }}{{/each}}</p>"` but it does happen if you give plain objects and arrays to the same stache.  I'll update the test to show this.